scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -152,3 +159,43 @@ def __iter__(self):
 
     def get_generated_keys(self) -> List[str]:
         return [self._name]
+
+
+class AttackPointIteratorRepeat(AttackPointIterator):
+    """
+    Attack point iterator class that iterates 
+    over a configuration a repeat amount of times.
+    """
+
+    def __init__(self,
+                 operation: str,
+                 configuration: Dict,
+                 repetitions: int = -1) -> None:
+        """Initialize the repeated iterate if repetitions is not present
```

Comment:
Much better, still no information about the possibility of raising an exception. Also not according to the style guide https://google.github.io/styleguide/pyguide.html#383-functions-and-methods (another comment had a linked example https://github.com/google/scaaml/blob/main/scaaml/capture/input_generators/input_generators.py#L44 )